### PR TITLE
Feature/feature dynamic script resolver support

### PR DESCRIPTION
**Pull request by badvision**
This evolution will not be televised.

### CODE REVIEW

**Comment by badvision** [comment]
@joerghoh, @davidjgonzalez  this introduces a necessary extension point to mcp, doesn't change any functionality otherwise.

File: bundle/src/main/java/com/adobe/acs/commons/mcp/DynamicScriptResolverService.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+/**
+ * Provides hooks for injecting additional MCP process definitions.
+ * For example, definitions that are not written in Java as OSGi services.
+ * Identifier is up to the resolver to determine, examples might be a unique name or a path.
+ * It is recommended to use something as specific as JCR path if at all possible for identifier.
+ */
+@ConsumerType
+public interface DynamicScriptResolverService {
+    Map<String, ProcessDefinitionFactory> getDetectedProcesDefinitionFactories(ResourceResolver rr);
```
Comment (joerghoh) [inline bundle/src/main/java/com/adobe/acs/commons/mcp/DynamicScriptResolverService.java]: Can you provide some javadoc for this publicly exposed interface and its methods?

File: bundle/src/main/java/com/adobe/acs/commons/mcp/impl/ControlledProcessManagerImpl.java [hunk 2]
```diff
@@ -64,6 +73,9 @@
     @Reference(cardinality = ReferenceCardinality.MANDATORY_MULTIPLE, bind = "bindDefinitionFactory", unbind = "unbindDefinitionFactory", referenceInterface = ProcessDefinitionFactory.class, policy = ReferencePolicy.DYNAMIC)
     private final List<ProcessDefinitionFactory> processDefinitionFactories = new CopyOnWriteArrayList<>();
 
+    @Reference(cardinality = ReferenceCardinality.OPTIONAL_MULTIPLE, bind = "bindScriptResolverService", unbind = "unbindScriptResolverService", referenceInterface = DynamicScriptResolverService.class, policy = ReferencePolicy.DYNAMIC)
```
Comment (joerghoh) [inline bundle/src/main/java/com/adobe/acs/commons/mcp/impl/ControlledProcessManagerImpl.java]: if the bind and unbind methods are as simple as here, I don't think that they are necessary. IIRC these implementations a kind of default (though I don't have the specific chapter of the OSGI spec handy).

File: bundle/src/main/java/com/adobe/acs/commons/mcp/DynamicScriptResolverService.java (hunk 1 above)
Comment (badvision) [reply bundle/src/main/java/com/adobe/acs/commons/mcp/DynamicScriptResolverService.java]: Sure thing!

File: bundle/src/main/java/com/adobe/acs/commons/mcp/impl/ControlledProcessManagerImpl.java (hunk 2 above)
Comment (badvision) [reply bundle/src/main/java/com/adobe/acs/commons/mcp/impl/ControlledProcessManagerImpl.java]: Yeah, I think so to but at some point during development I ran into a weird bug where optional services weren't binding and I needed these methods to debug it.  I think it's a hold-over but ultimately benign.

**Pull request merged**